include/librealsense2/dds/dds-device.cpp

```diff
@@ -43,7 +43,7 @@ std::shared_ptr< dds_device > dds_device::create( std::shared_ptr< dds_participa
                                                   topics::device_info const & info )
 {
     std::shared_ptr< dds_device > dev = find( guid );
-
+    // TODO a device may be created between find() and the next lock...
```

Comment:
Can be solved instead of TODO.
find() can receive a parameter need_to_lock = true, only create will call it with false.

Reply:
Not my intent with this PR -- I just wanted to point out a weakness.